### PR TITLE
Use Tempo's own API instead of Jira REST API

### PR DESCRIPTION
**Pull request by lunaticare**
* authenticate Tempo with API token, see ["Using the REST API as an application developer"](https://tempo-io.github.io/tempo-api-docs/)
* use Tempo's own [API](https://tempo-io.github.io/tempo-api-docs/#worklogs)

API authentication is useful if you use SSO login in JIRA.

### CODE REVIEW

**Comment by augsteyer** [comment]
Apologies for the delay, apparently I need to be watching my own repositories to get notifications for them.

I like the support for Cloud API, however, you are substituting the support of Private Server API for Cloud API here. And I kinda need the server support :) 
It would be better to keep both, and create a switch in the config file to use one or the other as they are not the same thing, see their [dev docs](https://www.tempo.io/developers).
e.g. `useTempoCloud: true/false`

Or, if you do not want to bother with extra work, you should adjust the logic in your fork and rename the project to identify your support for Tempo Cloud API only.

You would also want to run the linter to make sure some of the styling is adjusted, e.g.
```
cd [project]
npm i
./node_modules/.bin/eslint -c .eslintrc index.js helpers/ test/
```

File: index.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ const { tempoClient, togglClient } = require('./services/client.js')
 const { queryTogglEntries } = require('./services/togglEntries')
 const { queryTempoEntries } = require('./services/tempoEntries')
 
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
```
Comment (augsteyer) [inline index.js:15]: will need a docblock
e.g.
```
/**
 * A delay for tempo entry loading
 *
 * @param {number} ms
 * @return {Promise<number>}
 */
function sleep (ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}
```

File: index.js [hunk 2]
```diff
@@ -37,16 +41,17 @@ const transferFromTogglToTempo = async (from, to, utc, dryRun = false) => {
   // create worklogs in tempo from toggl time entries
   await bluebird.map(
     parsedEntries,
-    async ({ issueKey, start, duration, comment }) => (
+    async ({ issueKey, start, duration, comment }) => {
+      await sleep(Math.random() * 20000)
+      console.log(issueKey + ' ' + comment)
```
Comment (augsteyer) [inline index.js:46]: per linter requirements, template usage: 
```suggestion
      console.log(`${issueKey} ${comment}`)
```

File: index.js [hunk 3]
```diff
@@ -37,16 +41,17 @@ const transferFromTogglToTempo = async (from, to, utc, dryRun = false) => {
   // create worklogs in tempo from toggl time entries
   await bluebird.map(
     parsedEntries,
-    async ({ issueKey, start, duration, comment }) => (
+    async ({ issueKey, start, duration, comment }) => {
+      await sleep(Math.random() * 20000)
```
Comment (augsteyer) [inline index.js:45]: not quite sure why there is a random delay here

File: index.js [hunk 4]
```diff
@@ -37,16 +41,17 @@ const transferFromTogglToTempo = async (from, to, utc, dryRun = false) => {
...
+        authorAccountId: config.tempoWorker,
+        issueKey: issueKey,
+        startDate: formatDate(start),
         timeSpentSeconds: duration,
         billableSeconds: duration,
-        comment,
-        includeNonWorkingDays: true
-      }))
+        description: comment
+      })}
```
Comment (augsteyer) [inline index.js:54]: will need to adjust per linter styles as well:
```
      })
    }
```

**Review by augsteyer — CHANGES_REQUESTED** [review]

**Comment by lunaticare** [comment]
Will close the PR as I no longer use Tempo, and don't have access to test the changes.

**Pull request closed without merging**